### ISSUE TITLE
Speakername is editable even when edit function is disabled

### ISSUE THREAD

**Issue by murezzda** [bbc/react-transcript-editor#133]
**Describe the bug**
Setting `isEditable` to `false` does not prevent editing of speaker names.

**To Reproduce**
1. Set `isEditable` in the `TranscriptEditor` component to `false`.
2. Click on speaker label to edit the speaker.

**Expected behavior**
Speakers are also disabled from edits if `isEditable` is `false`

**Additional context**
If this change is implemented, maybe remove the edit-icon from the speaker.
![logo](https://user-images.githubusercontent.com/47388020/55707034-eae70380-59e2-11e9-8cbb-a726afb5df48.png)

**Comment by pietrop** [comment]
Good catch, as you might remember from PR https://github.com/bbc/react-transcript-editor/pull/103 in 

Leaving some notes on possible things to look at to address this. it might be a matter of 
- [ ] Passing the param `isEditable` from `TimedTextEditor` to `WrapperBlock`.
- [ ] adding a `if` in [`TimedTextEditor/WrapperBlock.js`](https://github.com/bbc/react-transcript-editor/blob/master/src/lib/TranscriptEditor/TimedTextEditor/WrapperBlock.js#L101) in the [`handleOnClickEdit`](https://github.com/bbc/react-transcript-editor/blob/master/src/lib/TranscriptEditor/TimedTextEditor/WrapperBlock.js#L32) function to check for he `isEditable` param.

and to remove the icon if not editable (good call) 
- [ ] In `WrapperBlock` [`speakerElement`](https://github.com/bbc/react-transcript-editor/blob/master/src/lib/TranscriptEditor/TimedTextEditor/WrapperBlock.js#L91) would need the `isEditable` param so that in the [`SpeakerLabel`](https://github.com/bbc/react-transcript-editor/blob/master/src/lib/TranscriptEditor/TimedTextEditor/SpeakerLabel.js) component it could know when to hide the speaker icon.

**Comment by RizwanSyed357** [comment]
Hey all, new contributor here.  Happy to try and help out with this.

Could someone please tell me exactly what the `isEditable` switch in the `TranscriptEditor` component looks like/how to access it in the UI? I would like to try it myself on your deployed app, but unfortunately looking at the `TranscriptEditor` component code is not enough to highlight where the `isEditable` switch is. Looking at the `TimedTextEditor` code isn't really helping either.

The only option I can see is in the UI is the `Text Is Editable`  checkbox, but that is not part of the `TranscriptEditor` component. 

Cheers

**Comment by pietrop** [comment]
Hi @RizwanSyed357 
Thanks for reaching out and welcome to the project!

The text is editable checkbox, is where you can toggle the [`isEditable `](https://github.com/bbc/react-transcript-editor/blob/master/demo/app.js#L259) attribute for `TranscriptEditor` [in the demo app](https://bbc.github.io/react-transcript-editor/?path=/story/demo--default) - (click load demo first)

this prop is being passed to [`TimedTextEditor` in `TranscriptEditor`](https://github.com/bbc/react-transcript-editor/blob/master/packages/components/transcript-editor/index.js#L344) - and then in [`TimedTextEditor` state](https://github.com/bbc/react-transcript-editor/blob/master/packages/components/timed-text-editor/index.js#L32)

At the moment to [`isEditable` attribute is implemented at line `#L104` of `TimedTextEditor` under the DraftJS `onChange` event listener](https://github.com/bbc/react-transcript-editor/blob/master/packages/components/timed-text-editor/index.js#L104)

if it's set to true then changes are saved if false then doesn't update state, making component read only.

---

This is how it is currently implemented, there might be room for a refactor using [draftJs `readOnly`](https://draftjs.org/docs/api-reference-editor#readonly) prop.

---

re more specifically with this issue of not allowing to edit speakers when` isEditable` is set to `false` for `TranscriptEditor` [these are some of the steps from the previous comment worth exploring](https://github.com/bbc/react-transcript-editor/issues/133#issuecomment-480766322)

let me know if you have any further questions or any of this is not clear, happy to help.

**Comment by RizwanSyed357** [comment]
Hi @pietrop ,

Thanks for this. I tried adding an if-condition to the `handleClickOnEdit` method in WrapperBlock.js. That didn't work.

I also added the `readOnly` prop to the `<Editor>` component, in the `<TimedTextEditor>`, but I can still change the Speakername after clicking on the icon. 

Here is the `readOnly` prop that I have added to the code: 

`<Editor
          editorState={ this.state.editorState }
          onChange={ this.onChange }
          stripPastedStyles
          blockRendererFn={ this.renderBlockWithTimecodes }
          handleKeyCommand={ command => this.handleKeyCommand(command) }
          keyBindingFn={ e => this.customKeyBindingFn(e) }
          spellCheck={ this.props.spellCheck }
          readOnly={ !this.props.isEditable }
        />`

This way when isEditable is true, readOnly is false - and vica versa. But it doesn't seem to be working. 

If you have any tips, that would be great. I am looking through the documentation to find out why my `readOnly` prop isn't working, but can't find anything.

**Comment by RizwanSyed357** [comment]
Hi,

I set the `readOnly` prop to `<Editor>` but unfortunately the speakername can still be changed when readOnly is set to true. 

And using an `if` condition   in the handleOnClickEdit of TimedTextEditor/WrapperBlock.js  doesn't work either - it doesn't update when the isEditable prop does. 

Would you like me to push my attempt to you so you can take a look? 

Riz

**Comment by pietrop** [comment]
Hi @RizwanSyed357, 

I tried [`readOnly`](https://draftjs.org/docs/api-reference-editor#readonly) that didn't seem to work, unless I am missing something from the draftJs docs.

But yes, feel free to raise a PR happy to have a look, and see if we can figure this out, I suspect there's something to do with propagating changes from `TimedTextEditor` to `WrapperBlock`
and the communication between those two components.

There's another PR where we have tried to address some of the inter component communication to reduce unnecessary re-renders https://github.com/bbc/react-transcript-editor/pull/160 which is still pending review, but might help.
